sigar_interface.go

```diff
@@ -18,6 +18,11 @@ type Sigar interface {
 
 var ErrNotImplemented error = errors.New("Collection not implemented for this operating system")
 
+// Simple Get() that returns an error
+type Getter interface {
+	Get() error
+}
```

Comment:
Careful @matthew-dailey, you might start wanting to use Go more often!

Ideally we would refer to all collectors via an interface like this, which would aid downstream users. While we can use `Getter` to call `Get()`, we wouldn't (yet) be able to extract the individual attributes. In the future, you might see another PR extending this concept to allow access to the struct attributes.